internal/cli/enterprise.go

```diff
@@ -82,6 +88,35 @@ func InitEnterpriseCLI(binaryName, version, dateBuilt string, schema *service.Co
 			rpLogger.SetStreamSummary(s)
 			return nil
 		}),
+
+		// Secrets management
+		service.CLIOptCustomRunFlags([]cli.Flag{
+			&cli.StringSliceFlag{
+				Name:  "secrets",
+				Usage: "Attempt to load secrets from a provided URN. If a referenced secret isn't identified using the provided method then environment variables will also be looked up, you can disable this behaviour with the --disable-env-lookup flag.",
+			},
+			&cli.BoolFlag{
+				Name:  "disable-env-lookup",
+				Usage: "Disable the ability for configs to interpolate environment variables with ${FOO} syntax",
+			},
```

Comment:
Should the env lookup just be the default for the above flag?

Reply:
You could have a "disabled" backend that does nothing for an easy fallback. Also support a base scheme or the empty scheme so it's a little more natural and you can do --secrets=disabled (or --secrets=none)

This is also fine too, I just generally prefer less knobs (it's harder to forget one that way, where a missing env var will explode noisily).

Reply:
Got a branch to try it out. The only awkwardness seems to be that if you want to override the default behaviour and disable env lookups you need to do something like `rpk connect run --secrets none: ./foo.yaml`. I think that's probably fine though as I doubt many people would want to do this.

Reply:
Yeah I think this works better, pushed the change and updated the description to match.